java/org/apache/catalina/connector/CoyoteAdapter.java

```diff
@@ -1252,6 +1252,11 @@ public static boolean checkNormalize(MessageBytes uriMB) {
 
         int pos = 0;
 
+        // An empty URL is not acceptable
+        if (start == end) {
+            return false;
+        }
+
         // Check for '\' and 0
```

Comment:
I believe the above return would represent unreachable code. Given this is on the code path every request must pass through, we do not want to add any unnecessary tests however trivial. If you can find an input to `CoyoteAdapter.postParseRequest()` that would result in this code being executed then that would likely be sufficient justification to add this test.

Reply:
I do not have an input that would trigger a 500 if we removed that check. I understand the performance consideration, so I'll remove it.